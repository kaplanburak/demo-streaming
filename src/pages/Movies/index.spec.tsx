import React from "react";
import { shallow } from "enzyme";
import { Movies } from "./index";

describe("Movies component", () => {
  const mockSetPageTitle = jest.fn();
  const mockGetMovies = jest.fn();

  it("should render", () => {
    let movies;
    beforeEach(() => {
      movies = shallow(
        <Movies
          setPageTitle={mockSetPageTitle}
          getMovies={mockGetMovies}
          list={[]}
          isLoading={false}
          gotError={false}
        />
      );
    });
    expect(movies).toMatchSnapshot();
  });

  it("should invoke functions once", () => {
    expect(mockSetPageTitle).toHaveBeenCalledWith("Popular Movies");
    expect(mockGetMovies.mock.calls.length).toBe(1);
  });
});
