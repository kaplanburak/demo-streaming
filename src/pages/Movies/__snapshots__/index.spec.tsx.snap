// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies component should render 1`] = `undefined`;
